service/history/mutableStateBuilder.go

```diff
@@ -3804,6 +4143,30 @@ func (e *mutableStateBuilder) syncActivityToReplicationTask(
 	)
 }
 
+func (e *mutableStateBuilder) updateWithLastWriteEvent(
+	lastEvent *workflow.HistoryEvent,
+) error {
+
+	e.GetExecutionInfo().LastEventTaskID = lastEvent.GetTaskId()
+
+	if e.replicationState != nil {
```

Comment:
Move nil check inside UpdateReplicationStateLastEventID?

Reply:
this is to avoid breaking all kinds of tests......